### PR TITLE
List schedule page

### LINKED ISSUE #9: Create list schedule page
# Context

Our users needs a schedule to follow their appointments.

Figma: https://www.figma.com/file/Rr0Fl9UNDeCUAAFHTvHi86/Iago-Project?node-id=5%3A312&t=DJ1YQ0Dj5BctjAUV-0

image

# Details
The initial page setup was implemented on #4 now we need to polish it to look the same as the design.

# Tech planning

1 - First we need to add the new types into the `types.ts`
```
// Types.ts
export type Appointment = {
    id: number;
    customerId: number;
    time: string;
};

export type Schedule = {
    [week: string]: Appointment[]
}
```

2- Let's create a new file to hold all of our mock data. Create a `src/constants.ts` file then let's add our mocked data (including the customers):
```
// constants.ts
export const CUSTOMERS: Customer[] = [
    {
        id: 1,
        customerName: 'Steve Lawrence',
        nickName: 'LSteve123'
    },
    {
        id: 2,
        customerName: 'Alex Andrade',
        nickName: 'AndradeAlex'
    }
];

export const SCHEDULE: Schedule = {
    '2023-04-07': [
        {
            id: 1,
            customerId: 1,
            time: '10:00'
        },
        {
            id: 2,
            customerId: 2,
            time: '11:30'
        }
    ],
    '2023-04-09': [
        {
            id: 3,
            customerId: 1,
            time: '07:00'
        },
        {
            id: 4,
            customerId: 2,
            time: '09:30'
        }
    ]
}
```

3 - Update the `CustomerPage.tsx` to get the mocked Customers from the `constants.ts` file:
```
import {CUSTOMERS} from `../constants`;


// Remove line 23 to 34 where you have the ITEMS
// Then on line 48 let's change to use the imported customers

// This is just an example for the items prop, keep all the other ones.
<DataTable items={CUSTOMERS} />
```

4 - Update the `SchedulePage.tsx` to import the mocked data:
```
import { SCHEDULE, CUSTOMERS } from '../constants';
```

** Challenge **
You need to update the `SchedulePage.tsx` to have one DataTable for each day in the `SCHEDULE` variable. 
TIP: You will have to iterate through the days, since that is an object and not an array, you will need to get the keys using [Object.keys()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys).

Also, in the `SCHEDULE` variable you only have the customer id, but you will need to show the name and nickname in the table. You will have to find the customer in the `CUSTOMERS` variable by the id to be able to display the name properly.

# Acceptance criteria

* there is a schedule for the users to follow their appointments. For now it's not functioning, just the UI.

### PR DESCRIPTION
**Pull request by IagoGoncalves22**
Resolves #9 

## Problem

We don't have a schedule page interface.

## Changes proposed in this PR

1. Created a file types.ts and add the new types we need to our mock data.
2. Created a src/constants.ts file and add our mocked data (including the customers).
3. Update the CustomerPage.tsx and customized the UI according to our design in Figma.
4. Created a file ScheduleRow.tsx as a component to our file SchedulePage.tsx.
5. Improved the internacionalization to our SchedulePage.tsx (us, pt, es).

### CODE REVIEW

File: src/components/CustomerRow.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ const CustomerRow = (props: DataTableCustomComponent<Customer>) => {
     const ACTIONS: DataTableAction[] = [
         {
             action: 'delete',
-            label: t('customerRow.delete'),
+            label: t('customerRow.cancel'),
```
Comment (galexandrade) [inline src/components/CustomerRow.tsx:19]: I think this action should still be `Delete`
image

File: src/components/ScheduleRow.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { Appointment, Customer } from '../types';
+
+type Item = Appointment & { customer: Customer };
+
+const ScheduleRow = (props: DataTableCustomComponent<Item>) => {
+    const [isOpen, setIsOpen] = useState(false);
+
+    const customer = props.item.customer;
+
+    const ACTIONS: DataTableAction[] = [
```
Comment (galexandrade) [inline src/components/ScheduleRow.tsx:24]: Nice. Great job 🎉

File: src/pages/SchedulePage.tsx [hunk 3]
```diff
@@ -1,43 +1,142 @@
...
+    ];
+
+    const dayOfWeek = daysOfWeek[date.getDay()];
+    const month = monthsOfYear[date.getMonth()];
+    const dayOfMonth = date.getDate();
+
+    const finalDateString = `${dayOfWeek}, ${month} ${dayOfMonth}`;
+
+    return finalDateString;
+};
```
Comment (galexandrade) [inline src/pages/SchedulePage.tsx:142]: We might need this format in other parts of the app. Let's move it into a file that we can reuse it later on:

- Create the `src/utils/date.ts` file
- Move this function into that file and export it.

```
export const organizeDate = (dateString: string) => {
    const date = new Date(dateString + 'T00:00');

    const daysOfWeek = [
        t('schedulePage.sunday'),
        t('schedulePage.monday'),
        t('schedulePage.tuesday'),
        t('schedulePage.wednesday'),
        t('schedulePage.thursday'),
        t('schedulePage.friday'),
        t('schedulePage.saturday')
    ];
    const monthsOfYear = [
        t('schedulePage.jan'),
        t('schedulePage.feb'),
        t('schedulePage.mar'),
        t('schedulePage.apr'),
        t('schedulePage.may'),
        t('schedulePage.jun'),
        t('schedulePage.jul'),
        t('schedulePage.aug'),
        t('schedulePage.sep'),
        t('schedulePage.oct'),
        t('schedulePage.nov'),
        t('schedulePage.dec')
    ];

    const dayOfWeek = daysOfWeek[date.getDay()];
    const month = monthsOfYear[date.getMonth()];
    const dayOfMonth = date.getDate();

    const finalDateString = `${dayOfWeek}, ${month} ${dayOfMonth}`;

    return finalDateString;
};
```

Then on this file you just need to remove this code block and import it from the utils file.

**Review by galexandrade — APPROVED** [review]
Great job @IagoGoncalves22 👏 
I just left minor suggestions. When you are done with them you can go ahead and merge this PR.

File: src/pages/SchedulePage.tsx [hunk 4]
```diff
@@ -1,43 +1,142 @@
...
-            name: 'date'
+            name: 'time',
+            label: t('schedulePage.time')
         },
+
         {
-            label: t('scheduledPage.hours'),
-            name: 'hours'
+            name: 'customer',
+            label: t('schedulePage.customer')
```
Comment (galexandrade) [inline src/pages/SchedulePage.tsx:32]: You can add a `size: 3` (the number might be another one) prop  so the columns looks like the style
image

File: src/components/CustomerRow.tsx (hunk 1 above)
Comment (IagoGoncalves22) [reply src/components/CustomerRow.tsx:19]: Alright. It was corrected.

File: src/components/ScheduleRow.tsx (hunk 2 above)
Comment (IagoGoncalves22) [reply src/components/ScheduleRow.tsx:24]: Thank you!

File: src/pages/SchedulePage.tsx (hunk 3 above)
Comment (IagoGoncalves22) [reply src/pages/SchedulePage.tsx:142]: Nice. It looks like too much better in this way.

File: src/pages/SchedulePage.tsx (hunk 4 above)
Comment (IagoGoncalves22) [reply src/pages/SchedulePage.tsx:32]: Great. It was done.

**Pull request merged**